### PR TITLE
Start creating the new overhauled UI

### PR DESCRIPTION
**Pull request by shocklateboy92**
dockui-1

This lays out the groundwork. There's more to be done on the C++ side, but I'd like to get this reviewed first. 

Also, in the meantime, we can start porting the rest of the views into modules.

### CODE REVIEW

**Comment by shocklateboy92** [comment]
I decided to use the same QML engine to render each module for now. This way, we can use a Singleton type to store the current attribute in QML, and the backend doesn't need to know anything about it - technically, it's presentation logic anyway.

Here's a screenshot of it retaining the correct 'currentAttribute' between different instances of the Ability Scores Module:

dockui-2

**Comment by shocklateboy92** [comment]
It now retains the position, visibility and dock status of each module. As detailed in 947b1f5ecbfa1315323f4f238e062c8b9a5e1eb2, it was really easy due to Qt's magic.

Here's a screenshot of regedit.exe showing the settings:

dockui-3

**Comment by shocklateboy92** [comment]
I ported the remaining modules, and implemented the Module resource requesting/matching system.

This is probably ready to be used in game now.

dockui-4

File: app/app-main.cpp [hunk 1]
```diff
@@ -7,14 +7,23 @@
...
 #include <src/project-context.h>
 #include "db.h"
+#include "ui-module.h"
+
+static const char * PB_UI_NAMESPACE = "org.lasath.psychic_bear.ui";
 
 int main(int argc, char *argv[])
 {
     QApplication app(argc, argv);
+    app.setOrganizationName("Lasath Foundation");
```
Comment (jarjee) [inline app/app-main.cpp:23]: Are you actually a foundation?

File: app/container-window.cpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    action->setText(module->name());
+    action->setCheckable(true);
+    views->addAction(action);
+
+    return action;
+}
+
+void ContainerWindow::setupUi()
+{
+    setWindowTitle("Hello World");
```
Comment (jarjee) [inline app/container-window.cpp:91]: Isn't this supposed to be Psychic Bear? You made it official in app-main.cpp

File: app/app-main.cpp (hunk 1 above)
Comment (shocklateboy92) [inline app/app-main.cpp:23]: Yes, thus why it's "lasath.org" rather than "lasath.com"

File: app/container-window.cpp (hunk 2 above)
Comment (shocklateboy92) [inline app/container-window.cpp:91]: Didn't want to add any regressions ;-)

**Pull request merged**